### ISSUE TITLE
remove notificationtrigger

### ISSUE THREAD

**Issue by mafer0410** [lordmilko/PrtgAPI#87]
Hi, I would like to delete notifications from a group of devices using the OnNotificationAction property, that is, delete those notifications that appear in the OnNotificationAction "WAN notification"

Additionally when one creates a state-type notification, and to edit its properties, how is it performed?

Get-device -tag vendors_Cisco | remove-notificationtrigger -rawparameters @{ OnNotification="Notificación LAN"}

I had done that command but it gives error.

**Comment by mafer0410** [comment]
Get-device -tag *vendors_Cisco* | remove-notificationtrigger -rawparameters @{ OnNotification="Notificación LAN"}


I had done that command but it gives error.

**Comment by lordmilko** [comment]
Hi @mafer0410,

In order to remove a notification trigger you need to first *retrieve* a notification trigger

```powershell
# Remove all triggers whose OnNotificationAction is "Notificación LAN"
Get-Device -Tag vendors_Cisco | Get-Trigger "Notificación LAN" | Remove-Trigger
```

To modify the properties of a notification trigger use the `Set-TriggerProperty` cmdlet

```powershell
# Set the latency of all state notification triggers applied to probes to 40 seconds
Get-Probe | Get-Trigger -Inherited $false -Type State | Set-TriggerProperty Latency 40
```

For more information please see the [wiki](https://github.com/lordmilko/PrtgAPI/wiki/Notification-Triggers)

Regards,
lordmilko

**Comment by mafer0410** [comment]
Hello, thank you very much, it worked as indicated, but I have only a
little doubt, you can edit and only look at the notifications of type
'Status', I have a notification of 'Speed' but when executing the command,
it gives error due to the notification of speed.

**Comment by lordmilko** [comment]
Hi @mafer0410,

What is the error you get exactly?

**Comment by mafer0410** [comment]
Hi, i have 2 errors,
1. the first error is treated when I want to enter any data and validate it
in the command, so that if the user enters any type of data, we indicate it
is an error and enter it again, but it does not accept numerical values.

**Comment by mafer0410** [comment]
The second error is as follows, an object has 2 status notifications and a speed notification, when wanting to make a query from the PowerShell where I want to list the notifications of such object (device) gives an error message.  It does not recognize notifications different from those of the state type.
![01](https://user-images.githubusercontent.com/51204835/60056304-b14a9e00-969d-11e9-851c-a54cf684f367.jpg)
![02](https://user-images.githubusercontent.com/51204835/60056305-b1e33480-969d-11e9-8215-1bdc99bfd544.jpg)
![03](https://user-images.githubusercontent.com/51204835/60056306-b1e33480-969d-11e9-8964-94d17bb35861.jpg)

**Comment by lordmilko** [comment]
In your first screenshot, the error you are getting clearly explains what the issue is. The `-Type` parameter expects a value of type [PrtgAPI.TriggerType](https://github.com/lordmilko/PrtgAPI/blob/master/PrtgAPI/Enums/Deserialization/Triggers/TriggerType.cs), of which the possible values are `State`, `Volume`, `Speed`, `Volume` or `Change`. You cannot use the Spanish words for these values, just like you can't type `Obtener-Disparador` instead of `Get-Trigger`. If you want to allow people to enter Spanish words, you will need to convert them prior to passing them to PrtgAPI.

The second error you are getting is a bit more tricky; the message is implying that one of your speed triggers is corrupt. This error occurs when PrtgAPI can't figure out what the channel referenced in your trigger refers to; in this case, the channel is called `Primario`. Retrieving notification triggers is not actually supported by PRTG, so PrtgAPI does its best to make sense of the information it can retrieve; unfortunately, this does not always work well in languages outside of English.

In this case, instead of returning the numeric ID of the primary channel (-999) PRTG simply returns the appropriate translation of the word "Primary" ("Primario" in Spanish) which PrtgAPI then tries to convert to a [StandardTriggerChannel](https://github.com/lordmilko/PrtgAPI/blob/master/PrtgAPI/Enums/Enhanced/TriggerChannel.cs#L313). While this works fine in English, users running non-English versions of PRTG will experience the error you are seeing.

I have discovered a potential solution for this issue, however I do not currently have an ETA as to when it will be implemented. In the meantime, the only way to get rid of this error is to manually remove any speed, threshold or volume triggers from your groups, devices and probes that refer to "standard" trigger values (Primary, Total, etc) rather than *specific* triggers defined on a sensor

I have opened #91 to track the progress of this issue

Regards,
lordmilko

**Comment by lordmilko** [comment]
Just to provide an update on this, I prototyped a potential solution for this today and discovered there's a lot more work that needs to be done in regards to Notification Triggers in general to make them work properly on non-English PRTG servers than just fixing up how channel names are deserialized. As such, at this stage these fixes won't likely be shipped until PrtgAPI 0.9.9 (after I release .NET/PowerShell Core support,, which is slated for 0.9.8. No current ETA on either).

In the meantime, the only full work around is to temporarily change your server's language to English, perform any tasks you may need to do, then change it back to your preferred language. This will only really suit however if you are doing a one off task; if you will need to be manipulating triggers constantly, this may not be an appropriate solution

Regards,
lordmilko

**Comment by mafer0410** [comment]
apply the language apply the language change in the Prtg and everything worked correctly.
[Final.txt](https://github.com/lordmilko/PrtgAPI/files/3332020/Final.txt)
change in the Prtg and everything worked correctly.
